coordinator/src/main/java/org/apache/uniffle/coordinator/SimpleClusterManager.java

```diff
@@ -179,47 +185,105 @@ public void nodesCheckTest() {
     nodesCheck();
   }
 
-  private void updateExcludeNodes(String path) {
-    int originalExcludeNodesNumber = excludeNodes.size();
+  private synchronized void updateExcludeNodes(String path) {
+    int originalExcludeNodesNumber = excludedNodes.size();
     try {
       Path hadoopPath = new Path(path);
       FileStatus fileStatus = hadoopFileSystem.getFileStatus(hadoopPath);
       if (fileStatus != null && fileStatus.isFile()) {
         long latestModificationTime = fileStatus.getModificationTime();
         if (excludeLastModify.get() != latestModificationTime) {
-          parseExcludeNodesFile(hadoopFileSystem.open(hadoopPath));
+          excludedNodes = parseExcludedNodesFile(hadoopFileSystem.open(hadoopPath), true);
+          LOG.info(
+              "Updated exclude nodes and {} nodes were marked as exclude nodes",
+              excludedNodes.size());
+          // update exclude nodes and last modify time
           excludeLastModify.set(latestModificationTime);
         }
       } else {
-        excludeNodes = Sets.newConcurrentHashSet();
+        excludedNodes = Sets.newConcurrentHashSet();
       }
     } catch (FileNotFoundException fileNotFoundException) {
-      excludeNodes = Sets.newConcurrentHashSet();
+      excludedNodes = Sets.newConcurrentHashSet();
     } catch (Exception e) {
       LOG.warn("Error when updating exclude nodes, the exclude nodes file path: {}.", path, e);
     }
-    int newlyExcludeNodesNumber = excludeNodes.size();
+    int newlyExcludeNodesNumber = excludedNodes.size();
     if (newlyExcludeNodesNumber != originalExcludeNodesNumber) {
-      LOG.info("Exclude nodes number: {}, nodes list: {}", newlyExcludeNodesNumber, excludeNodes);
+      LOG.info("Exclude nodes number: {}, nodes list: {}", newlyExcludeNodesNumber, excludedNodes);
     }
-    CoordinatorMetrics.gaugeExcludeServerNum.set(excludeNodes.size());
+    CoordinatorMetrics.gaugeExcludeServerNum.set(excludedNodes.size());
   }
 
-  private void parseExcludeNodesFile(DataInputStream fsDataInputStream) throws IOException {
+  private Set<String> parseExcludedNodesFile(
+      DataInputStream fsDataInputStream, boolean notNeedComment) throws IOException {
     Set<String> nodes = Sets.newConcurrentHashSet();
     try (BufferedReader br =
         new BufferedReader(new InputStreamReader(fsDataInputStream, StandardCharsets.UTF_8))) {
       String line;
       while ((line = br.readLine()) != null) {
-        if (!StringUtils.isEmpty(line) && !line.trim().startsWith("#")) {
+        if (notNeedComment) {
```

Comment:
I think we don't need this condition. We don't need to read comments in both cases.